### ISSUE TITLE
Add an unmanaged version of the pool

### ISSUE THREAD

**Issue by bikeshedder** [bikeshedder/deadpool#25]
@svenknobloch proposed in #24 to add an `add` function to the pool which allows using the pool where objects are not generated during runtime but upfront. This version of the pool does not need a manager and timeouts as no creation and/or recycling needs to happen.

**Comment by bikeshedder** [comment]
I just added the implementation for that "passive" `Pool` to the repository. See https://github.com/bikeshedder/deadpool/blob/master/src/passive.rs

### Usage

```rust
use deadpool::passive::Pool;
#[tokio::main]
async fn main() {
    let pool = Pool::from_vec(vec![
        "foo".to_string(),
        "bar".to_string(),
    ]);
    {
        let s = pool.get().await;
        assert_eq!(s.len(), 3);
        assert_eq!(pool.status().available, 1)
    }
    assert_eq!(pool.status().available, 2);
}
```

The code is a lot simpler and the API of it is way better suited for fixed-size pools that don't require a manager to create/recycle/destroy objects as needed.

@svenknobloch Could you please check if this code is practical for your usecase?

All that's missing are some unit tests to ensure the correctness of it and it's ready to be pushed to `crates.io`.

If anyone has a suggestion for a better name I'm open for suggestions. My initial thought was `static` but having to write `r#static` in an public API didn't feel right. Other name candidates were `fixed` and `constant_sized` until I decided to put the bikeshedding aside and just call it `passive` for the time being.

**Comment by svenknobloch** [comment]
Yep that works for me!

For the naming, I would name the module static internally and then exporting the name for the public api as `StaticPool` from the root, but that is of course up to you.

I also think it would make more sense to change `from_vec` to something like `from_iter` and have it take an `ExactSizeIterator` so that it's a bit more generic and can be used for more than just vecs. (I can PR this real quick if you want)

**Comment by bikeshedder** [comment]
PRs are always welcome. 👍

Regarding naming things: I'm currently thinking of renaming the current "default" pool into `managed` and the other one into `unmanaged` as their main difference is the lack of a `Manager` for the unmanaged one. I would move the current pool into the `deadpool::managed` module then and keep a type alias in the crate root for backwards compatibility.

**Comment by svenknobloch** [comment]
That seems like a fair solution to me.

**Comment by bikeshedder** [comment]
This is now implemented and will soon be released as part of `deadpool 0.5`.
